queries.sql

```diff
@@ -29,3 +27,80 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with weights that equal precisely 10.4kg or 17.3kg):
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+-- Verify the change (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that the species column went back to the original state (optional)
+SELECT * FROM animals;
+
+
+BEGIN;
+
+-- Update species to 'digimon' for animals whose name ends in 'mon'
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update species to 'pokemon' for animals without a species set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+-- Verify the changes (optional)
+SELECT * FROM animals;
+
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Verify that records were deleted (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that all records in the animals table still exist (optional)
+SELECT * FROM animals;
+
+-- Inside a transaction:
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in 'mon'.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-- Verify that changes were made.
+BEGIN;
+
+-- Update species to 'digimon' for animals whose name ends in 'mon'
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update species to 'pokemon' for animals without a species set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+-- Verify the changes (optional)
+SELECT * FROM animals;
+
+-- Answering the additional questions
+
+-- How many animals are there?
+SELECT COUNT(*) AS total_animals FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS never_escaped FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts FROM animals GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS average_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```

Comment:
- [ ] As suggested by the previous reviewer, kindly consider adding the screenshots for these queries in your pull request.